pkg/virt-launcher/virtwrap/network/network.go

```diff
@@ -97,7 +97,7 @@ func getPodInterfaceName(networks map[string]*v1.Network, cniNetworks map[string
 	}
 }
 
-func SetupNetworkInterfacesPhase1(vmi *v1.VirtualMachineInstance, pid int) error {
+func mockableSetupPodNetworkPhase1(vmi *v1.VirtualMachineInstance, pid int) error {
```

Comment:
The name hints its usage (to be mocked) and not what it does.
Having in the production code a function name "mockable" is too odd IMO.

Converting it to a private method sounds more appropriate to me.
On the other hand, removing the globals should be the primary goal, but I do recognize it requires much more work.

Reply:
The `mockable` function name is very odd, but the problem is not with its name. The problem is that it exists only because we want to mock it in the tests. This PR wants to put these oddities on the surface, so every reader of the code immediately understands why this function exist. For me, it took few hours and few phonecalls to understand that this function exists in production only because the test wants it. I would like that future readers would not have this ordeal again.

The name keeps a hit about what it does `SetupPodNetworkPhase1`, though one would argue that `Phase1` is an opaque term. This PR does not try to fix everything, only few of them, one at a time.

Reply:
You will indeed expose the mess, but on the other hand I am unsure if new ones who read the code (even me), will really understand why we have a `mockable` function.
I suggest to add this info through a comment.

Reply:
You forced me to look deeper into the code, and I find no reason to have this mocking at all. I'll push a branch which drops it.